django/test/selenium.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import unicode_literals
+
+import sys
+import unittest
+
+from django.test import LiveServerTestCase, tag
+from django.utils.module_loading import import_string
+from django.utils.six import with_metaclass
+from django.utils.text import capfirst
+
+
+class SeleniumTestCaseBase(type(LiveServerTestCase)):
+    # List of browsers to dynamically create test classes for.
+    browsers = []
+    # Sentinel value to differentiate browser-specific instances.
+    browser = None
+
+    def __new__(cls, name, bases, attrs):
+        """
+        Dynamically create new classes and add them to the test module when
+        multiple browsers specs are provided (e.g. --selenium=firefox,chrome).
+        """
+        test_class = super(SeleniumTestCaseBase, cls).__new__(cls, name, bases, attrs)
+        # If the test class is either browser-specific or a test base, return it.
+        if test_class.browser or not any(name.startswith('test') and callable(value) for name, value in attrs.items()):
+            return test_class
+        elif test_class.browsers:
+            # Reuse the created test class to make it browser-specific.
+            # We can't rename it to include the browser name or create a
+            # subclass like we do with the remaining browsers as it would
+            # either duplicate tests or prevent pickling of its instances.
+            first_browser = test_class.browsers[0]
+            test_class.browser = first_browser
+            # Create subclasses for each of the remaining browsers and expose
+            # them through the test's module namespace.
+            module = sys.modules[test_class.__module__]
+            for browser in test_class.browsers[1:]:
+                browser_test_class = cls.__new__(
+                    cls,
+                    str("%s%s" % (capfirst(browser), name)),
+                    (test_class,),
+                    {'browser': browser, '__module__': test_class.__module__}
+                )
+                setattr(module, browser_test_class.__name__, browser_test_class)
+            return test_class
+        # If no browsers were specified skip all the tests declared on this class.
+        return unittest.skip('No browsers specified.')(test_class)
```

Comment:
@timgraham, let me know if the additional comments help you get your head around what's going on here.